src/model/CoilHeatingElectricMultiStageStageData.cpp

```diff
@@ -0,0 +1,227 @@
+      boost::optional<double> value = getDouble(OS_Coil_Heating_Electric_MultiStage_StageDataFields::Efficiency, true);
+      OS_ASSERT(value);
+      return value.get();
+    }
+
+    boost::optional<double> CoilHeatingElectricMultiStageStageData_Impl::nominalCapacity() const {
+      return getDouble(OS_Coil_Heating_Electric_MultiStage_StageDataFields::NominalCapacity, true);
+    }
+
+    bool CoilHeatingElectricMultiStageStageData_Impl::isNominalCapacityAutosized() const {
+      bool result = false;
+      boost::optional<std::string> value = getString(OS_Coil_Heating_Electric_MultiStage_StageDataFields::NominalCapacity, true);
+      if (value) {
+        result = openstudio::istringEqual(value.get(), "autosize");
+      }
+      return result;
+    }
+
+    bool CoilHeatingElectricMultiStageStageData_Impl::setEfficiency(double Efficiency) {
+      bool result = setDouble(OS_Coil_Heating_Electric_MultiStage_StageDataFields::Efficiency, Efficiency);
+      return result;
+    }
+
+    bool CoilHeatingElectricMultiStageStageData_Impl::setNominalCapacity(boost::optional<double> NominalCapacity) {
+      bool result(false);
+      if (NominalCapacity) {
+        result = setDouble(OS_Coil_Heating_Electric_MultiStage_StageDataFields::NominalCapacity, NominalCapacity.get());
+      }
+      return result;
+    }
+
+    void CoilHeatingElectricMultiStageStageData_Impl::autosizeNominalCapacity() {
+      bool result = setString(OS_Coil_Heating_Electric_MultiStage_StageDataFields::NominalCapacity, "autosize");
+      OS_ASSERT(result);
+    }
+
+    boost::optional<std::tuple<int, CoilHeatingElectricMultiStage>> CoilHeatingElectricMultiStageStageData_Impl::stageIndexAndParentCoil() const {
+
+      boost::optional<std::tuple<int, CoilHeatingElectricMultiStage>> result;
+
+      // This coil performance object can only be found in a CoilHeatingElectricMultiStage
+      // Check all CoilHeatingElectricMultiStages in the model, seeing if this is inside of one of them.
+      boost::optional<int> stageIndex;
+      boost::optional<CoilHeatingElectricMultiStage> parentCoil;
+      auto coilHeatingElectricMultiStages = this->model().getConcreteModelObjects<CoilHeatingElectricMultiStage>();
+      for (const auto& coilInModel : coilHeatingElectricMultiStages) {
+        // Check the coil performance objects in this coil to see if one of them is this object
+        std::vector<CoilHeatingElectricMultiStageStageData> perfStages = coilInModel.stages();
+        int i = 1;
+        for (auto perfStage : perfStages) {
+          if (perfStage.handle() == this->handle()) {
+            stageIndex = i;
+            parentCoil = coilInModel;
+            break;
+          }
+          i++;
+        }
+      }
+
+      // Warn if this coil performance object was not found inside a coil
+      if (!parentCoil) {
+        LOG(Warn, name().get() + " was not found inside a CoilCoolingDXMultiSpeed in the model, cannot retrieve the autosized value.");
```

Comment:
Copy paste error. And this is probably the wrong message in the initial file actually... this isn't the autosized method